app/assets/javascripts/service_request.coffee

```diff
@@ -0,0 +1,119 @@
+setEquipment = ->
+  equipments = $('.select_location option:selected').data('equipments')
+  $('.select_equipment').find('option').remove()
+  $('.select_equipment').append $('<option></option>')
+  $('.existing-equipment').addClass('hidden')
+  $.each equipments, (i, e) ->
+    dataEquipment = JSON.stringify e
+    $('.select_equipment').append $('<option></option>').attr({'value': e.id, 'data-equipment': dataEquipment }).text(e.id)
+
+$('.select_location').livequery ->
+  setEquipment()
+
+$(document).on 'change', '.select_location', ->
+  setEquipment()
+  
+setEquipmentFields = ->
+  equipment = $('.select_equipment option:selected').data('equipment')
+  $('.new-equipment').addClass('hidden')
+  $('.equipment-warranty').addClass('hidden')
+  if equipment
+    $.ajax
```

Comment:
Can you please take a look at `select2`? Maybe it will be easier and required less code? I know there are simple solutions with this to update one select, after you update value of previous select, here is examples: http://stackoverflow.com/questions/13268083/select2-changing-items-dynamically